type.go

```diff
@@ -519,6 +598,24 @@ func (t byteArrayType) Decode(dst encoding.Values, src []byte, enc encoding.Enco
 	return encoding.DecodeByteArray(dst, src, enc)
 }
 
+func (t byteArrayType) AssignValue(dst reflect.Value, src Value) error {
+	v := src.ByteArray()
+	switch dst.Kind() {
+	case reflect.String:
+		dst.SetString(string(v))
+		return nil
+	case reflect.Slice:
+		if dst.Type().Elem().Kind() == reflect.Uint8 {
+			dst.SetBytes(copyBytes(v))
+			return nil
+		}
```

Comment:
How about not checking so it panics if the underlying type is not compatible, which would also be the same behavior if we reached the code below?
```suggestion
		dst.SetBytes(copyBytes(v))
		return nil
```